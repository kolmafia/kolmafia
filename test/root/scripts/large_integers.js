const {print, toString: kolmafiaToString} = require("kolmafia");

[
  11,
  11.0,
  2147483647, // Java Integer.MAX_VALUE
  2147483648, // Java Integer.MAX_VALUE + 1
  2147483648.0,
  11_111_111_111,
  9007199254740991, // JavaScript Number.MAX_SAFE_INTEGER
].forEach((value) => {
  const asString = String(value);
  const asKolmafiaString = kolmafiaToString(value);
  const delimitedString = kolmafiaToString(value, "%,d");
  print(JSON.stringify({
    value,
    isSafeInteger: Number.isSafeInteger(value),
    asString,
    asKolmafiaString,
    stringsEqual: asString === asKolmafiaString,
    delimitedString,
  }));
});

print("");

[
  9007199254740992, // JavaScript Number.MAX_SAFE_INTEGER + 1, this will not be converted to a long, but to a double
  11.5, // this will be a double
  1.1, // this will be a double
  2147483648.5, // this will be a double
].forEach((value) => {
  const asString = String(value);
  const asKolmafiaString = kolmafiaToString(value);
  print(JSON.stringify({
    value,
    isSafeInteger: Number.isSafeInteger(value),
    asString,
    asKolmafiaString,
    stringsEqual: asString === asKolmafiaString,
  }));
});