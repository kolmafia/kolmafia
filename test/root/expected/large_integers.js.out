{"value":11,"isSafeInteger":true,"asString":"11","asKolmafiaString":"11","stringsEqual":true,"delimitedString":"11"}
{"value":11,"isSafeInteger":true,"asString":"11","asKolmafiaString":"11","stringsEqual":true,"delimitedString":"11"}
{"value":2147483647,"isSafeInteger":true,"asString":"2147483647","asKolmafiaString":"2147483647","stringsEqual":true,"delimitedString":"2,147,483,647"}
{"value":2147483648,"isSafeInteger":true,"asString":"2147483648","asKolmafiaString":"2147483648","stringsEqual":true,"delimitedString":"2,147,483,648"}
{"value":2147483648,"isSafeInteger":true,"asString":"2147483648","asKolmafiaString":"2147483648","stringsEqual":true,"delimitedString":"2,147,483,648"}
{"value":11111111111,"isSafeInteger":true,"asString":"11111111111","asKolmafiaString":"11111111111","stringsEqual":true,"delimitedString":"11,111,111,111"}
{"value":9007199254740991,"isSafeInteger":true,"asString":"9007199254740991","asKolmafiaString":"9007199254740991","stringsEqual":true,"delimitedString":"9,007,199,254,740,991"}

{"value":9007199254740992,"isSafeInteger":false,"asString":"9007199254740992","asKolmafiaString":"9007199254740992.0","stringsEqual":false}
{"value":11.5,"isSafeInteger":false,"asString":"11.5","asKolmafiaString":"11.5","stringsEqual":true}
{"value":1.1,"isSafeInteger":false,"asString":"1.1","asKolmafiaString":"1.1","stringsEqual":true}
{"value":2147483648.5,"isSafeInteger":false,"asString":"2147483648.5","asKolmafiaString":"2147483648.5","stringsEqual":true}